File: solutions/javascript/raindrops/1/raindrops.js
//
// This is only a SKELETON file for the 'Raindrops' exercise. It's been provided as a
// convenience to get you started writing code faster.
//
function convertWithTable(n, sounds){
  const result = sounds
    .filter(sound => n % sound.key === 0)
    .map(sound => sound.sound)
    .join("")
  return result.length > 0 ? result : `${n}`
}


export const convert = (n) => {
  return convertWithTable(n, [{key: 3, sound: "Pling"}, {key: 5, sound: "Plang"},{key: 7, sound: "Plong"}])
};

File: solutions/go/raindrops/1/raindrops.go
package raindrops

import "strconv"

func Convert(number int) string {
    sounds := make (map[int]string)
    sounds[3] = "Pling"
    sounds[5] = "Plang"
    sounds[7] = "Plong"
	return ConvertWithSounds(number, sounds)
}

func ConvertWithSounds(number int, sounds map[int]string) string {
    res := ""
    for n, sound := range sounds{
        if (number % n == 0) {
            res += sound;
        }
    }
    if (res == "") {
        return strconv.Itoa(number)
    } else {
    	return res;
    }
}